backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java

```diff
@@ -12,54 +12,157 @@
+                    licensePara.setSpacingAfter(0);
+                    addHyperLink(licensePara, "bookmark_" + licenseName, licenseName);
+                }
+            } else {
+                XWPFRun errorRun = document.createParagraph().createRun();
+                String errorText = nullToEmptyString(parsingResult.getMessage());
+                String filename = getFilename(parsingResult);
+                addFormattedText(errorRun, String.format("Error reading license information: %s", errorText), FONT_SIZE, false, ALERT_COLOR);
+                addFormattedText(errorRun, String.format("Source file: %s", filename), FONT_SIZE, false, ALERT_COLOR);
+            }
+            addNewLines(document.createParagraph().createRun(), 1);
+        }
+        addPageBreak(document.createParagraph().createRun());
+    }
 
-        addLicenseTextsHeader(document, "Appendix - License Texts");
-        addLicenseTexts(document, projectLicenseInfoResults);
+    private void addReleaseTitle(XWPFDocument document, LicenseInfoParsingResult parsingResult) {
+        String releaseTitle = nullToEmptyString(parsingResult.getName()) + " " + nullToEmptyString(parsingResult.getVersion());
+        XWPFParagraph releaseTitleParagraph = document.createParagraph();
+        releaseTitleParagraph.setStyle(STYLE_HEADING);
+        addBookmark(releaseTitleParagraph, releaseTitle, releaseTitle);
+        addNewLines(document.createParagraph().createRun(), 0);
     }
-}
 
+    private Set<String> getReleaseCopyrights(LicenseInfoParsingResult licenseInfoParsingResult) {
+        Set<String> copyrights = Collections.emptySet();
+        if (licenseInfoParsingResult.isSetLicenseInfo()) {
+            LicenseInfo licenseInfo = licenseInfoParsingResult.getLicenseInfo();
+            if (licenseInfo.isSetCopyrights()) {
+                copyrights = licenseInfo.getCopyrights();
+            }
+        }
+        return copyrights;
+    }
+
+    private Set<String> getReleasesLicenses(LicenseInfoParsingResult licenseInfoParsingResult) {
+        Set<String> licenses = new HashSet<>();
+        if (licenseInfoParsingResult.isSetLicenseInfo()) {
+            LicenseInfo licenseInfo = licenseInfoParsingResult.getLicenseInfo();
+            if (licenseInfo.isSetLicenseNamesWithTexts()) {
+                for (LicenseNameWithText licenseNameWithText : licenseInfo.getLicenseNamesWithTexts()) {
+                    licenses.add(licenseNameWithText.isSetLicenseName()
+                            ? licenseNameWithText.getLicenseName()
+                            : UNKNOWN_LICENSE_NAME);
+                }
+            }
+        }
+        return licenses;
+    }
+
+    private String getFilename(LicenseInfoParsingResult licenseInfoParsingResult) {
+        return Optional.ofNullable(licenseInfoParsingResult.getLicenseInfo())
+                .map(LicenseInfo::getFilenames)
+                .flatMap(l -> l.stream().findFirst())
```

Comment:
Why not take all instead of only the first one?

Reply:
@alexbrdn is there a special reason why we should only take the first file name?

Reply:
I figured out that only the first item in the stream occurs the parsing failure.
All previous files should be fine.